### DISCUSSION TITLE
cannot update row

### DISCUSSION THREAD

**Discussion by revl94** [adonisjs/core#2257] [General]
Package version:
"@adonisjs/lucid": "^6.1.3"

Node.js and npm version
node: v14.15.1
npm: v6.14.8

Sample Code:
I try to do this:

const user = await User.findByOrFail('email', email)

  user.firstName = firstName
  user.lastName = lastName
  user.country = country
  user.countryCode = countryCode
  user.phoneNumber = phoneNumber

  await user.save()
  
  return user
im using pg for database, when execute this method dont update the row.
Plis help!

**Comment by revl94** [comment]
{
  method: 'first',
  options: {},
  timeout: false,
  cancelOnTimeout: false,
  bindings: [ 'rvielma94@gmail.com', 1 ],
  __knexQueryUid: '3b84e495-e634-4727-aa50-53d0177f4b2a',
  sql: 'select * from "users" where "email" = ? limit ?'
}
{
  method: 'update',
  options: {},
  timeout: false,
  cancelOnTimeout: false,
  bindings: [
    '44e32718-2abb-4717-846b-7b5953fc4a08',
    'Lima',
    'Colombia',
    'CO',
    '123456789',
    '2021-02-08 08:58:00',
    '2021-02-13 15:01:17',
    '44e32718-2abb-4717-846b-7b5953fc4a08'
  ],
  __knexQueryUid: 'cc50d89f-0850-4030-8e78-5b7a297df665',
  sql: 'update "users" set "id" = ?, "lastName" = ?, "country" = ?, "countryCode" = ?, "phoneNumber" = ?, "created_at" = ?, "updated_at" = ? where "id" = ?',
  returning: undefined
}

debbug

**Reply by urishabh11** [reply]
@revl94 share schema of User

**Reply by revl94** [reply]
this is the mutation:
type Mutation {
  updateUser( firstName: String!, lastName: String!, country: String!, countryCode: String!, phoneNumber: String!, email: String! ): User
}

and the shema:

this.create('users', (table) => {
      table.uuid('id').primary().unique()
      table.string('firstName', 80).notNullable()
      table.string('lastName', 80).notNullable()
      table.string('email', 254).notNullable().unique()
      table.string('country', 80).notNullable()
      table.string('countryCode', 80).notNullable()
      table.string('phoneNumber', 80).notNullable()
      table.string('state', 80).nullable()
      table.string('stateCode', 80).nullable()
      table.string('password', 60).notNullable()
      table.timestamps()
    })

@urishabh11

**Comment by revl94** [comment]
@urishabh11  i found the mistake, i had a hook "beforeSave" than generate a uuid.. and that hook need to be "beforecreate" because is only for create. Thank you anyway!!
